packages/menu/src/Menu.ts

```diff
@@ -190,35 +213,33 @@ export class Menu extends SpectrumElement {
                 ? 'none'
                 : ((this.getAttribute('role') || undefined) as RoleType);
             this.resolvedSelects = this.selects;
-            event.currentAncestorWithSelects = this;
         } else {
             this.resolvedRole = this.ignore
                 ? 'none'
                 : ((this.getAttribute('role') || undefined) as RoleType);
             this.resolvedSelects =
                 this.resolvedRole === 'none' ? 'ignore' : 'none';
         }
-    }
 
-    /**
-     * When a descendant `<sp-menu-item>` element is added or updated it will dispatch
-     * this event to announce its presence in the DOM. During the bubble phase the first
-     * Menu based element that the event encounters that does not inherit selection will
-     * manage the selection state of the dispatching `<sp-menu-item>` element.
-     * @param event
-     */
-    private onSelectableItemAddedOrUpdated(
-        event: MenuItemAddedOrUpdatedEvent
-    ): void {
         const selects =
             this.resolvedSelects === 'single' ||
             this.resolvedSelects === 'multiple';
+        event.item.menuData.cleanupSteps.push((item: MenuItem) =>
+            this.removeChildItem(item)
+        );
         if (
             (selects || (!this.selects && this.resolvedSelects !== 'ignore')) &&
             !event.item.menuData.selectionRoot
         ) {
             event.item.setRole(this.childRole);
-            event.selectionRoot = this;
+            event.item.menuData.selectionRoot =
+                event.item.menuData.selectionRoot || this;
+            if (event.item.selected) {
```

Comment:
I noticed that the submenu items aren't maintaining their selection state once the menu is reopened, like in [this example](https://overlay-mixins--spectrum-web-components.netlify.app/storybook/index.html?path=/story/menu-submenu--submenu). Is this logic potentially not saving the selection state, or is the expected behaviour supposed to be that the selection doesn't persist? This is how it worked in the previous version, if I recall, though, so it's not a huge deal if it isn't fixed. Just wanted to mention it. It seems like unexpected behaviour to me.

Reply:
Or, I guess my question would be, if we want the selected item to persist, does this put us on a good path forward to address that in the future?

Reply:
The story you linked here does not _ever_ have selection state, as it's an Action Menu.

[This story](https://overlay-mixins--spectrum-web-components.netlify.app/storybook/index.html?path=/story/menu-submenu--context-menu) has selection state, but the story is not written in such a way as to persist it (previously, too). I'll look into what would be needed so we could do that as a proof case.